### PR TITLE
Pycharm inspections

### PR DESCRIPTION
**Pull request by renemilk**
Changes inspired by pycharm's code inspections. I'll add more till the end of the week.

### CODE REVIEW

File: src/pymor/operators/ei.py [hunk 1]
```diff
@@ -144,7 +143,7 @@ def jacobian(self, U, mu=None):
         else:
             U_components = NumpyVectorArray(U.components(self.source_dofs), copy=False)
             JU = self.restricted_operator.jacobian(U_components, mu=mu) \
-                                         .apply(NumpyVectorArray(np.eye(len(self.source_dofs)), copy=False))
+                .apply(NumpyVectorArray(np.eye(len(self.source_dofs)), copy=False))
```
Comment (sdrave) [inline src/pymor/operators/ei.py]: If find, the original version is more readable.

File: src/pymor/playground/vectorarrays/disk.py [hunk 2]
```diff
@@ -44,8 +48,8 @@ class DiskVectorArray(VectorArrayInterface):
     def __init__(self, vectors, subtype=_NONE):
         if isinstance(vectors, ListVectorArray):
             assert subtype is self._NONE \
-                or (type(subtype) is tuple and len(subtype) == 2
-                    and subtype[0] == vectors.vector_type and subtype[1] == vectors.subtype)
+                   or (type(subtype) is tuple and len(subtype) == 2
```
Comment (sdrave) [inline src/pymor/playground/vectorarrays/disk.py]: Again, I would prefer the original version.

File: src/pymor/playground/vectorarrays/disk.py [hunk 3]
```diff
@@ -218,12 +222,12 @@ def almost_equal(self, other, ind=None, o_ind=None, rtol=None, atol=None):
...
             assert len(ind) == len(o_ind)
             return np.array([self._load(i).almost_equal(other._load(oi), rtol=rtol, atol=atol)
-                            for i, oi in izip(ind, o_ind)])
+                             for i, oi in izip(ind, o_ind)])
 
     def scal(self, alpha, ind=None):
         assert self.check_ind_unique(ind)
         assert isinstance(alpha, Number) \
-            or isinstance(alpha, np.ndarray) and alpha.shape == (self.len_ind(ind),)
+               or isinstance(alpha, np.ndarray) and alpha.shape == (self.len_ind(ind),)
```
Comment (sdrave) [inline src/pymor/playground/vectorarrays/disk.py]: see above

File: src/pymor/playground/vectorarrays/disk.py [hunk 4]
```diff
@@ -242,7 +246,7 @@ def axpy(self, alpha, x, ind=None, x_ind=None):
         assert self.space == x.space
         assert self.len_ind(ind) == x.len_ind(x_ind) or x.len_ind(x_ind) == 1
         assert isinstance(alpha, Number) \
-            or isinstance(alpha, np.ndarray) and alpha.shape == (self.len_ind(ind),)
+               or isinstance(alpha, np.ndarray) and alpha.shape == (self.len_ind(ind),)
```
Comment (sdrave) [inline src/pymor/playground/vectorarrays/disk.py]: see above

File: src/pymor/playground/vectorarrays/disk.py [hunk 5]
```diff
@@ -343,14 +347,14 @@ def l2_norm(self, ind=None):
     def components(self, component_indices, ind=None):
         assert self.check_ind(ind)
         assert isinstance(component_indices, list) and (len(component_indices) == 0 or min(component_indices) >= 0) \
-            or (isinstance(component_indices, np.ndarray) and component_indices.ndim == 1
-                and (len(component_indices) == 0 or np.min(component_indices) >= 0))
+               or (isinstance(component_indices, np.ndarray) and component_indices.ndim == 1
+                   and (len(component_indices) == 0 or np.min(component_indices) >= 0))
         ind = list(xrange(self._len)) if ind is None else [ind] if isinstance(ind, Number) else ind
```
Comment (sdrave) [inline src/pymor/playground/vectorarrays/disk.py]: see above

File: src/pymor/playground/vectorarrays/disk.py [hunk 6]
```diff
@@ -343,14 +347,14 @@ def l2_norm(self, ind=None):
...
 
         if len(ind) == 0:
             assert len(component_indices) == 0 \
-                or isinstance(component_indices, list) and max(component_indices) < self.dim \
-                or isinstance(component_indices, np.ndarray) and np.max(component_indices) < self.dim
+                   or isinstance(component_indices, list) and max(component_indices) < self.dim \
+                   or isinstance(component_indices, np.ndarray) and np.max(component_indices) < self.dim
             return np.empty((0, len(component_indices)))
```
Comment (sdrave) [inline src/pymor/playground/vectorarrays/disk.py]: see above

File: src/pymortests/vectorarray.py [hunk 7]
```diff
@@ -1081,7 +1082,8 @@ def test_almost_equal_wrong_ind(compatible_vector_array_pair):
     v1, v2 = compatible_vector_array_pair
     for ind1, ind2 in invalid_ind_pairs(v1, v2):
         if (ind1 is None and len(v1) == 1 or isinstance(ind1, Number) or hasattr(ind1, '__len__') and len(ind1) == 1 or
-            ind2 is None and len(v2) == 1 or isinstance(ind2, Number) or hasattr(ind2, '__len__') and len(ind2) == 1):  # NOQA
+                        ind2 is None and len(v2) == 1 or isinstance(ind2, Number) or hasattr(ind2, '__len__') and len(
+                ind2) == 1):  # NOQA
             continue
```
Comment (sdrave) [inline src/pymortests/vectorarray.py]: The new formatting is a mess ...

**Comment by sdrave** [comment]
@renemilk, most changes look nice! Could you update to current master and have a look at the changes I commented on?

**Comment by renemilk** [comment]
Updated. Will look at individual comments next.

**Comment by renemilk** [comment]
Any idea about that test failure, @sdrave? Also, forgot to ref this PR in 7a86cbe35b79ea0fb8a465f5df1f4875df3b583d

File: src/pymor/operators/fv.py [hunk 8]
```diff
@@ -165,7 +165,7 @@ def __init__(self, flux, flux_derivative, gausspoints=5, intervals=1):
         self.intervals = intervals
         self.build_parameter_type(inherits=(flux, flux_derivative))
         points, weights = GaussQuadratures.quadrature(npoints=self.gausspoints)
-        points = points / intervals
+        points /= intervals
```
Comment (sdrave) [inline src/pymor/operators/fv.py]: points is write-protected. The original version makes a copy of points, the new in-place version will fail.

File: src/pymor/playground/vectorarrays/disk.py [hunk 9]
```diff
@@ -202,6 +206,24 @@ def replace(self, other, ind=None, o_ind=None, remove_from_other=False):
...
+            a = self._load(ind[0])
+            return np.array([a.almost_equal(other._load(oi), rtol=rtol, atol=atol) for oi in o_ind])
+        elif len(o_ind) == 1:
+            b = other._load(o_ind[0])
+            return np.array([self._load(i).almost_equal(b, rtol=rtol, atol=atol) for i in ind])
+        else:
+            assert len(ind) == len(o_ind)
+            return np.array([self._load(i).almost_equal(other._load(oi), rtol=rtol, atol=atol)
+                            for i, oi in izip(ind, o_ind)])
+
```
Comment (sdrave) [inline src/pymor/playground/vectorarrays/disk.py]: almost_equal has been removed in master

File: src/pymortests/pickle.py [hunk 10]
```diff
@@ -84,7 +86,8 @@ def _assert_is_equal(first, second):
         elif not isinstance(first, BasicInterface):
             assert first == second
         else:
-            assert (set(first.__dict__.keys()) - ignored_attributes) == (set(second.__dict__.keys()) - ignored_attributes)
+            assert (set(first.__dict__.keys()) - ignored_attributes) == (
+                set(second.__dict__.keys()) - ignored_attributes)
```
Comment (sdrave) [inline src/pymortests/pickle.py]: I would say, the version with the too long line is more readable.

File: src/pymortests/vectorarray.py [hunk 11]
```diff
@@ -4,23 +4,23 @@
```
Comment (sdrave) [inline src/pymortests/vectorarray.py]: Most changes in this file revert changes made in master ...

**Comment by renemilk** [comment]
Yay me and my conflict resolutions I guess.

**Comment by sdrave** [comment]
Hmm, I believe it would be wise to run pycharm again on current master? Shall we close this one?

**Comment by renemilk** [comment]
Agreed.

**Pull request closed without merging**